test/test_other.py

```diff
@@ -3365,12 +3365,16 @@ def test_jspi_add_function(self):
   def test_embind_tsgen(self, opts):
     # Check that TypeScript generation works and that the program is runs as
     # expected.
-    self.do_runf('other/embind_tsgen.cpp', 'main ran',
-                 emcc_args=['-lembind', '--emit-tsd', 'embind_tsgen.d.ts'] + opts)
+    self.emcc(test_file('other/embind_tsgen.cpp'),
+              ['-o', 'embind_tsgen.mjs', '-lembind', '--emit-tsd', 'embind_tsgen.d.ts'] + opts)
 
     # Test that the output compiles with a TS file that uses the defintions.
-    cmd = shared.get_npm_cmd('tsc') + ['embind_tsgen.d.ts', '--noEmit']
+    shutil.copyfile(test_file('other/embind_tsgen.ts'), 'main.ts')
+    # A package file with type=module is needed to allow top level await in TS.
+    shutil.copyfile(test_file('other/embind_tsgen_package.json'), 'package.json')
+    cmd = shared.get_npm_cmd('tsc') + ['embind_tsgen.d.ts', 'main.ts', '--module', 'NodeNext', '--moduleResolution', 'nodenext']
```

Comment:
Why doesn't tsc generate a file called `.mjs` here I wonder?

Is the package.json file here needed for running `tsc` or for running the output?

Reply:
> Why doesn't tsc generate a file called `.mjs` here I wonder?

They prefer the package approach. IIRC the different extension was somewhat controversial.

> Is the package.json file here needed for running `tsc` or for running the output?

It's also needed for node to run it as an ES module. I've updated the comment.